{
	"name": "CVE-2021-29533",
	"type": "denial of service",
	"severity": "Low",
	"vector": "Local",
	"description": "A security issue has been found in TensorFlow before version 2.4.2. An attacker can trigger a denial of service via a `CHECK` failure by passing an empty image to `tf.raw_ops.DrawBoundingBoxes`. This is because the implementation(https://github.com/tensorflow/tensorflow/blob/ea34a18dc3f5c8d80a40ccca1404f343b5d55f91/tensorflow/core/kernels/image/draw_bounding_box_op.cc#L148-L165) uses `CHECK_*` assertions instead of `OP_REQUIRES` to validate user controlled inputs. Whereas `OP_REQUIRES` allows returning an error condition back to the user, the `CHECK_*` macros result in a crash if the condition is false, similar to `assert`. In this case, `height` is 0 from the `images` input. This results in `max_box_row_clamp` being negative and the assertion being falsified, followed by aborting program execution.",
	"groups": [
		"AVG-1962"
	],
	"packages": [
		"tensorflow"
	],
	"advisories": [],
	"references": [
		"https://github.com/tensorflow/tensorflow/security/advisories/GHSA-393f-2jr3-cp69",
		"https://github.com/tensorflow/tensorflow/commit/b432a38fe0e1b4b904a6c222cbce794c39703e87"
	],
	"notes": null
}